src/V3Width.cpp

```diff
@@ -3298,6 +3298,26 @@ class WidthVisitor final : public AstNVisitor {
                     relinkHandle.relink(patp);
                 }
             }
+
+            std::vector<std::pair<AstNodeDType*, AstPatMember*>> type_defaults;
+            for (AstPatMember* patp = VN_CAST(nodep->itemsp(), PatMember); patp;
+                 ) {
+                if (patp->isTypedDefault()) {
+                    AstNodeDType* type = VN_CAST(patp->keyp(), NodeDType);
+                    auto matching_type_it
+                        = find_if(type_defaults.begin(), type_defaults.end(),
+                                  [&type](auto x) { return type->similarDType(x.first); });
+                    if (matching_type_it == type_defaults.end()) {
+                        type_defaults.push_back({type, patp});
+                    }
```

Comment:
Is it legal to have the same type default twice?  Perhaps we should have an error?  (And a test for the error)?

Reply:
Yeah, I think that it's legal. Not from the perspective of IEEE, but in the [test linked with the issue](https://symbiflow.github.io/sv-tests-results/#Verilator|5.10|structure) there is this use case.